src/cleanlab_tlm/utils/chat_completions.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+from typing import TYPE_CHECKING, Any, Optional, cast
+
+import aiohttp
+
+from cleanlab_tlm.internal.api import api
+from cleanlab_tlm.internal.base import BaseTLM
+from cleanlab_tlm.internal.constants import (
+    _DEFAULT_TLM_QUALITY_PRESET,
+    _VALID_TLM_QUALITY_PRESETS_CHAT_COMPLETIONS,
+)
+from cleanlab_tlm.internal.types import TLMQualityPreset
+from cleanlab_tlm.internal.validation_openai import validate_openai_kwargs
+from cleanlab_tlm.tlm import TLM, TLMOptions, TLMScore
+from cleanlab_tlm.utils.chat import form_prompt_string
+
+if TYPE_CHECKING:
+    from openai.types.chat import ChatCompletion
+
+
+class TLMChatCompletion(BaseTLM):
+
+    def __init__(
+        self,
+        quality_preset: TLMQualityPreset = _DEFAULT_TLM_QUALITY_PRESET,
+        *,
+        api_key: Optional[str] = None,
+        options: Optional[TLMOptions] = None,
+        timeout: Optional[float] = None,
+    ):
+        super().__init__(
+            quality_preset=quality_preset,
+            valid_quality_presets=_VALID_TLM_QUALITY_PRESETS_CHAT_COMPLETIONS,
+            support_custom_eval_criteria=False,
+            api_key=api_key,
+            options=options,
+            timeout=timeout,
+            verbose=False,
+        )
+
+        self._tlm = TLM(
+            quality_preset=quality_preset,
+            api_key=api_key,
+            options=options,
+            timeout=timeout,
+        )
+
+    def create(
+        self,
+        **openai_kwargs: Any,
+    ) -> "ChatCompletion":
+        validate_openai_kwargs(**openai_kwargs)
+
+        return self._event_loop.run_until_complete(
+            self._create_async(openai_kwargs),
+        )
+
+    def score(
+        self,
+        response: "ChatCompletion",
+        **openai_kwargs: Any,
+    ) -> TLMScore:
+
+        if (messages := openai_kwargs.get("messages")) is None:
+            raise ValueError("messages is a required OpenAI input argument.")
+        tools = openai_kwargs.get("tools", None)
+
+        prompt_text = form_prompt_string(messages, tools)
+        response_text = _get_string_response(response)
```

Comment:
can you save a ticket for future PR:

we should try to extract token-probs out of `response` object if they are in there and include them in the score, like in TLMLite

Reply:
when the user does:  TLMChatCompletions.create()

the resulting scores should also be including token-probs (can save for future PR too)